.github/workflows/lint.yml

```diff
@@ -1,21 +1,28 @@
 name: lint
 on:
   workflow_dispatch:
+  pull_request:
   push:
+    branches:
+      - develop
 
 jobs:
   pre-commit:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
-      - uses: actions/setup-python@v2
+      - uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
```

Comment:
just curious, why do we need this github.event.... line now? what is it for?

Reply:
Couple weeks ago I changed this from running on `pull_request` to running on `push`. The intention was to have pre-commit scan the code on your branch (default `push` event behavior), instead of the resulting code of merging your branch into develop (default `pull_request` event behavior). But `push` scans weren't running for external contributors.

So now I switched back to running on `pull_request`, but manually forcing github to lint your branch instead of the PR merge result. This is helpful because if someone accidentally force-merges pre-commit failures to develop, because normally those failures would be silently merged into your PRs whenever CI runs on them, thereby failing your PRs even tho you didn't do anything wrong.